app/src/pages/_app.tsx

```diff
@@ -1,7 +1,26 @@
+import { useState } from 'react';
 import '../styles/globals.css'
 import type { AppProps } from 'next/app'
 
+
+
 function App({ Component, pageProps }: AppProps) {
-  return <Component {...pageProps} />
+  const [isDarkMode, setIsDarkMode] = useState(false);
+
+  return (
+    <>
+      {!isDarkMode && <div className="light-mode">
+        <div className="mode-change-box">
+          Change to {isDarkMode ? 'light' : 'dark'} mode <input type="checkbox" checked={isDarkMode} onChange={() => setIsDarkMode(!isDarkMode)} />
```

Comment:
```suggestion
          Change to {isDarkMode ? 'light' : 'dark'} mode <input type="checkbox" checked={isDarkMode} onChange={() => setIsDarkMode(prevMode => !prevMode)} />
```
前の状態変数の値を参照して次の状態変数の値を設定する場合は、dispatcher (`setState`的なやつ)の引数をcallbackにすると良いです！
理由はcallbackの引数はそのdispatcherが呼ばれるときの状態変数を参照しているため、常に最新の状態変数を参照できるからです！